fabric-dimension-patches-v1/src/main/java/net/fabricmc/fabric/mixin/MixinChangeDimensionFix.java

```diff
@@ -0,0 +1,42 @@
+package net.fabricmc.fabric.mixin;
+
+import net.minecraft.entity.Entity;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.server.world.ServerWorld;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.Heightmap;
+import net.minecraft.world.PortalForcer;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.HashSet;
+
+@Mixin(value = PortalForcer.class, priority = -100000)
+public class MixinChangeDimensionFix
+{
+	@Shadow
+	@Final
+	private ServerWorld world;
+
+	@Inject(at = @At("HEAD"), method = "usePortal", cancellable = true)
+	private void usePortal(final Entity entity, final float yaw, final CallbackInfoReturnable<Boolean> info) {
+		if(entity.method_5656() == null) {
```

Comment:
Not sure if I like this. all this does is check if the Vec3 field_6020 is null. The moment some modder just copies vanilla's code and accidentaly sets field_6020 to some value this fix would no longer work.
Have you considred just checking the destination dimension namespace and trigger the fix if it's not a minecraft dimension?

Reply:
Or maybe combine the null check with the dimension namespace using or to handle cases like traveling to the nether from a custom dimension without a nether portal.

Reply:
If a modder copies vanilla's code and sets field_6020 in their `changeDimension` mixin, then this patch would not be run (so, it's working properly)-- don't think it's much of an issue.

Maybe it would be better to remove the patch and just add in a simple API instead? It could be as simple as a `<DimensionType, TeleportAction>` map, where the `DimensionType` is the dimension you're registering teleportation mechanics for, and `TeleportAction` is an interface where you can define teleportation functionality (eg. we're going to the overworld, do xyz).

Reply:
Is field_6020 the only field that causes a NPE if it's not set? Was sure there was another on.

And an API is definetly a must have feature, but I don't see why this couldn't be used as a fallback fix if the modder forgets to use said API.